addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductsWidgetControlPanel.js

```diff
@@ -53,7 +53,11 @@ odoo.define('point_of_sale.ProductsWidgetControlPanel', function(require) {
                 let ProductIds = await this.rpc({
                     model: 'product.product',
                     method: 'search',
-                    args: [['&', ['name', 'ilike', this.searchWordInput.el.value + "%"], ['available_in_pos', '=', true]]],
+                    args: [['&',['available_in_pos', '=', true], '|','|','|',
+                     ['name', 'ilike', this.searchWordInput.el.value],
+                     ['default_code', 'ilike', this.searchWordInput.el.value],
+                     ['product_variant_ids.default_code', 'ilike', this.searchWordInput.el.value],
```

Comment:
Why are you searching on 'product_variant_ids.default_code' ? Because product_variant_ids is a one2Many to product.product and you are already search default_code on the product.product object

Reply:
Yes, it makes sense. I'll remove it. I didn't notice that the model was `product.product` and not `product.template`